src/vm/memory/relocatable.zig

```diff
@@ -96,7 +96,7 @@ pub const Relocatable = struct {
     /// This method fails if `self` and other` are not from the same segment.
     pub fn sub(self: Self, other: Self) !Self {
```

Comment:
Here, as a result of your refactor, can we enforce a better typing of the output?

Reply:
@tcoratger good point I will add a todo for that

Reply:
ah there's more comments, OK let's wait for Dhruv to fix it